src/composables/permissions.ts

```diff
@@ -0,0 +1,138 @@
+  {},
+  STORAGE_KEYS.permissions,
+  {
+    backgroundSync: true,
+    migrations: [
+      migratePermissionsVuexToComposable,
+    ],
+  },
+);
+
+export function usePermissions() {
+  function setTransactionSignLimitLeft(host: string, value: number) {
+    permissions.value[host] = {
+      ...permissions.value[host],
+      transactionSignLimitLeft: value,
+    };
+  }
+
+  function resetTransactionSignLimitLeft(host: string) {
+    permissions.value[host] = {
+      ...permissions.value[host],
+      transactionSignLimitLeft: permissions.value[host].transactionSignLimit,
+      transactionSignFirstAskedOn: (new Date()).toISOString(),
+    };
+  }
+
+  function addPermission(permission: IPermission) {
+    permissions.value[permission.host] = permission;
+  }
+
+  function removePermission(host: string) {
+    delete permissions.value[host];
+  }
+
+  function checkTransactionSignPermission(
+    host: string,
+    { amount = 0, fee = 0, nameFee = 0 }: ITransactionSignPermissionOptions,
+  ): boolean {
+    const {
+      transactionSignLimit,
+      transactionSignLimitLeft,
+      transactionSignFirstAskedOn,
+    } = permissions.value[host] || {};
+    const limitLeft = transactionSignLimitLeft - +aettosToAe(+amount + +fee + +nameFee);
+
+    if (!transactionSignLimit) {
+      return false;
+    }
+    if (
+      !transactionSignFirstAskedOn
+      || (
+        new Date().getTime() - new Date(transactionSignFirstAskedOn).getTime()
+        >= 24 * 60 * 60 * 1000
+      )
+    ) {
+      resetTransactionSignLimitLeft(host);
+    }
+    if (limitLeft < 0) {
+      return false;
+    }
+
+    setTransactionSignLimitLeft(host, limitLeft);
+    return true;
+  }
+
+  function checkPermissions(
+    host: string,
+    method: METHODS,
+    options: ITransactionSignPermissionOptions = {},
+  ): boolean {
+    const permission = permissions.value[host];
+    if (!permission) {
+      return false;
+    }
+    switch (method) {
+      case METHODS.sign: return checkTransactionSignPermission(host, options);
+      case METHODS.connect: return permission.address;
+      case METHODS.address: return permission.addressList;
+      case METHODS.signMessage: return permission.messageSign;
+      case METHODS.subscribeAddress: return permission.address;
+      default: return false;
+    }
+  }
+
+  async function requestAddressForHost(
+    host: string,
+    connectionPopupCallback: () => Promise<any>,
+  ) {
+    if (checkPermissions(host, METHODS.subscribeAddress)) {
+      return true;
+    }
+    try {
+      await connectionPopupCallback();
+      return true;
+    } catch {
+      return false;
+    }
+  }
+
+  async function requestAllAddressesForHost(
+    host: string,
+    connectionPopupCallback: () => Promise<any>,
+  ) {
+    if (checkPermissions(host, METHODS.address)) {
+      return true;
+    }
+    try {
+      await connectionPopupCallback();
+      return true;
+    } catch {
+      return false;
+    }
+  }
```

Comment:
Those two methods are exactly the same - maybe you could create base function for them and call it with different parameter?

Reply:
Yes, I refactored it. Please take a look again.